lib/devices_test.go

```diff
@@ -0,0 +1,197 @@
+package usb_tree_lib
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/google/gousb"
+)
+
+func TestDescToDevice(t *testing.T) {
+	desc := mockDesc()
```

Comment:
You're immediately calling a function I've not seen, try to sort the code in a logical order so that functions call functions defined above them whenever reasonably possible.